code/datums/components/battery_power.dm

```diff
@@ -0,0 +1,51 @@
+/// Generalized framework allowing something to draw power from an inserted power cell.
+/// Contains attack_hand() and attackby() signals allowing cells to be added and removed by hand.
+/// This base type supports only standard cells; device cells won't fit.
+/datum/component/battery_power
+	var/obj/item/cell/cell
```

Comment:
![image](https://imgs.xkcd.com/comics/standards.png
I think the alternative would be to have the cell on /obj, which potentially complicates objects that use _device_ cells, which iirc are a completely distinct type.

Reply:
I felt like it'd be unclean to have an obj-level variable even though not everything uses cells. would a component (or extension) not make the interactions cleaner?